reactos/dll/win32/browseui/basebarsite.cpp

```diff
@@ -472,10 +711,155 @@ LRESULT CBaseBarSite::OnNotify(UINT uMsg, WPARAM wParam, LPARAM lParam, BOOL &bH
+                HFONT newFont, oldFont;
+
+                index = SendMessage(RB_IDTOINDEX, fCurrentActiveBar->fBandID , 0);
+                ZeroMemory(&info, sizeof(info));
+                ZeroMemory(wszTitle, sizeof(wszTitle));
+                DrawEdge(pnmcd->hdc, &pnmcd->rc, EDGE_ETCHED, BF_BOTTOM);
+                // We also resize our close button
+                ::SetWindowPos(toolbarWnd, HWND_TOP, pnmcd->rc.right - 22, 0, 20, 18, SWP_SHOWWINDOW);
+                // Draw the text
+                info.cch = MAX_PATH;
+                info.lpText = wszTitle;
+                rt = pnmcd->rc;
+                rt.right -= 24;
+                rt.left += 2;
+                if (FAILED_UNEXPECTEDLY(GetInternalBandInfo(index, &info, RBBIM_TEXT)))
+                    return CDRF_SKIPDEFAULT;
+                newFont = GetTitleFont();
+                if (newFont)
+                    oldFont = (HFONT)SelectObject(pnmcd->hdc, newFont);
+                DrawText(pnmcd->hdc, info.lpText, -1, &rt, DT_SINGLELINE | DT_LEFT);
+                SelectObject(pnmcd->hdc, oldFont);
+                DeleteObject(newFont);
+                return CDRF_SKIPDEFAULT;
+            }
+            else
+            {
+                DrawEdge(pnmcd->hdc, &pnmcd->rc, EDGE_ETCHED, BF_BOTTOM);
+                // We also resize our close button
+                ::SetWindowPos(toolbarWnd, HWND_TOP, 0, 2, 20, 18, SWP_SHOWWINDOW);
+            }
+            return CDRF_SKIPDEFAULT;
+        default:
+            break;
+    }
+    return CDRF_DODEFAULT;
+}
+
+HFONT CBaseBarSite::GetTitleFont()
+{
+    NONCLIENTMETRICS mt;
+    mt.cbSize = sizeof(mt);
+    if (!SystemParametersInfo(SPI_GETNONCLIENTMETRICS, sizeof(mt), &mt, 0))
+    {
+        ERR("Can't get system parameters !\n");
+        return NULL;
+    }
+    return CreateFontIndirect(&mt.lfMenuFont);
+
+}
+
+HRESULT CBaseBarSite::FindBandByGUID(REFGUID pGuid, DWORD *pdwBandID)
+{
+    DWORD                       numBands;
+    DWORD                       i;
+    HRESULT                     hr;
+    REBARBANDINFO               bandInfo;
+    CBarInfo                    *realInfo;
+
+    hr = EnumBands(-1, &numBands);
+    if (FAILED_UNEXPECTEDLY(hr))
+        return E_FAIL;
+
+    for(i = 0; i < numBands; i++)
+    {
+        if (FAILED_UNEXPECTEDLY(GetInternalBandInfo(i, &bandInfo)))
+            return E_FAIL;
+        realInfo = (CBarInfo*)bandInfo.lParam;
+        if (IsEqualGUID(pGuid, realInfo->fBarClass))
+        {
+            *pdwBandID = realInfo->fBandID;
+            return S_OK;
+        }
+    }
+    return S_FALSE;
+}
+
+HRESULT CBaseBarSite::ShowBand(DWORD dwBandID)
+{
+    UINT                        index;
+    CComPtr<IDeskBand>          dockingWindow;
+    HRESULT                     hResult;
+    REBARBANDINFO               bandInfo;
+
+    // show our band
+    hResult = GetBandObject(dwBandID, IID_PPV_ARG(IDeskBand, &dockingWindow));
+    if (!SUCCEEDED(hResult))
```

Comment:
`FAILED_UNEXPECTEDLY` ?
Also, you used `{}` around the return, which you omit a few lines later.